src/lib/commands/create.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  ExtensionContext,
+  commands,
+  FileCreateEvent,
+  WorkspaceEdit,
+  Uri,
+  workspace,
+} from 'vscode';
+import { COMMANDS } from '../constants';
+import { getUniquePathName } from '../fileHelpers';
+const findup = require('findup-sync');
+const { createTemplate } = require('@hubspot/cli-lib/templates');
+const { createModule } = require('@hubspot/cli-lib/modules');
+const { createFunction } = require('@hubspot/cli-lib/functions');
+import * as path from 'path';
+
+// returns a listener that deletes the created file and then allows a given callback to run
+// doAfter receives unique filepath with the given extension
+// ex: extension given is "html", user types "example", but "example.html" already exists. doAfter receives "example1.html"
+const buildFileCreateListener = (
+  extension: string,
+  folderPath: string,
+  doAfter: Function,
+  cleanupCallback: Function
+) => {
+  return async (e: FileCreateEvent) => {
+    if (
+      e.files?.length === 1 &&
+      new RegExp(`${folderPath}/.*`).test(e.files[0].fsPath)
+    ) {
+      cleanupCallback();
+      const uniqueFilePath = getUniquePathName(e.files[0].fsPath, extension);
+      if (e.files[0].fsPath !== uniqueFilePath) {
+        const edit = new WorkspaceEdit();
+        edit.renameFile(e.files[0], Uri.file(uniqueFilePath));
+        await workspace.applyEdit(edit);
+      }
+      doAfter(uniqueFilePath);
+    }
+  };
+};
+
+// doAfter receives (filepath: string)
+const onClickCreateFile = (extension: string, doAfter: Function) => {
+  return (clickContext: any) => {
+    if (clickContext.scheme === 'file') {
+      const createFileSubscription = workspace.onDidCreateFiles(
+        buildFileCreateListener(extension, clickContext.fsPath, doAfter, () =>
+          createFileSubscription.dispose()
+        )
+      );
+      commands.executeCommand('explorer.newFile');
+    }
+  };
+};
+const onClickCreateFolder = (extension: string, doAfter: Function) => {
+  return (clickContext: any) => {
+    if (clickContext.scheme === 'file') {
+      const createFileSubscription = workspace.onDidCreateFiles(
+        buildFileCreateListener(extension, clickContext.fsPath, doAfter, () =>
+          createFileSubscription.dispose()
+        )
+      );
+      commands.executeCommand('explorer.newFolder');
+    }
+  };
+};
+
+export const registerCommands = (context: ExtensionContext) => {
+  context.subscriptions.push(
+    commands.registerCommand(
+      COMMANDS.CREATE.SECTION,
+      onClickCreateFile('html', (uniqueFilePath: string) => {
+        createTemplate(
+          path.basename(uniqueFilePath),
+          path.dirname(uniqueFilePath),
+          'section',
```

Comment:
`'section'` here refers to `section.html` in https://github.com/HubSpot/hubspot-cli/tree/create-serverless-functions/packages/cli-lib/defaults

and likewise for other template creation commands

Reply:
Could we create constants for these file types?